k8s/grafana/README.md

```diff
@@ -8,6 +8,36 @@ data collection systems - including Prometheus, InfluxDB, Elasticsearch, MySQL o
 
 Popular open stacks on Kubernetes packaged by Google.
 
+## Design
+
+![Architecture diagram](resources/grafana-architecture.png)
+
+This application offers a vanilla deployment of stateful Grafana installation
+on a Kubernetes cluster.
+
+Administrative user credentials are automatically configured in the application
+through a Kubernetes Secret. Configuration file (/etc/grafana/grafana.ini) is
+defined in a ConfigMap and mounted to Grafana StatefulSet.
+
+By default, the Service exposing Grafana server is of type ClusterIP, which makes
+it accessible only in a private network.
+
+## Configuration
+
+**Fluentd** - it contains a ConfigMap defining a few files for logs locations
+  and formats. It exports logs to Elasticsearch daily indexes, starting with
+  “logstash-\*” prefix.
+
+**Elasticsearch** - forms a cluster with a configurable number of replicas
+  (specified by user before the installation); It uses a dedicated disk
+  (VolumeClaim) for storing data in a stateful manner.
+
+**Kibana** - it stores all the configuration in Elasticsearch index, this is
+  why Kibana itself is installed in a single-replica, “stateless” Deployment.
+  Kibana has initial configuration represented by JSON payloads that are used
+  against Kibana’s REST API to set an index pattern for “logstash-\*” and some
+  useful saved searches.
+
```

Comment:
Do we need info about Fluentd, ElasticSearch and Kibana here?

Reply:
Nope - removed.